aws/code/rapids_cloud_ml.py

```diff
@@ -15,289 +15,450 @@
+    #  parse ML model parameters [ e.g., passed in by cloud HPO ]
+    # -------------------------------------------------------------------------------------------------------------
+    def parse_hyper_parameter_inputs ( self, input_args, model_type, compute_type ):
+        print('parsing model hyper-parameters from command line arguments...\n')
+        parser = argparse.ArgumentParser ()
+
+        if 'XGBoost' in model_type:
+            parser.add_argument( '--max_depth',       type = int,   default = default_max_depth )
+            parser.add_argument( '--num_boost_round', type = int,   default = default_n_estimators )
+            parser.add_argument( '--subsample',       type = float, default = .25 )
+            parser.add_argument( '--learning_rate',   type = float, default = 0.3 )            
+            parser.add_argument( '--lambda_l2',       type = float, default = .2 )            
+            parser.add_argument( '--gamma',           type = float, default = 0. )            
+            parser.add_argument( '--alpha',           type = float, default = 0. )
+            
+            args, unknown_args = parser.parse_known_args( input_args )
             
-        elif 'multi-CPU' in self.compute_type:
-            self.n_workers = os.cpu_count()
-            self.cluster = LocalCluster(  n_workers = self.n_workers, threads_per_worker = 1 )
-            self.client = Client( self.cluster )
-            print(f'dask multi-CPU cluster with {self.n_workers} workers')
+            model_params = {            
+                'max_depth' : args.max_depth,
+                'num_boost_round': args.num_boost_round,
+                'learning_rate': args.learning_rate,
+                'gamma': args.gamma,
+                'lambda': args.lambda_l2,
+                'random_state' : 0,
+                'verbosity' : 0,
+                'objective' : 'binary:logistic'
+            }        
+
+            if 'single-CPU' in compute_type:
+                model_params.update( { 'nthreads': os.cpu_count() })
+
+            if 'GPU' in compute_type:
+                model_params.update( { 'tree_method': 'gpu_hist' })
+            else:
+                model_params.update( { 'tree_method': 'hist' })
+            
+        elif 'RandomForest' in model_type:
+            parser.add_argument( '--max_depth',    type = int,   default = default_max_depth )
+            parser.add_argument( '--n_estimators', type = int,   default = default_n_estimators )            
+            parser.add_argument( '--max_features', type = float, default = .25 )
+
+            args, unknown_args = parser.parse_known_args( input_args )
+
+            model_params = {            
+                'max_depth' : args.max_depth,
+                'n_estimators' : args.n_estimators,        
+                'max_features': args.max_features,
+                'seed' : 0,
```

Comment:
Would be nice to allow users to pass in a fixed seed.

Reply:
Done